robot/Cumulus/tests/browser/recurring_donations/create_recurring_donation_ui.robot

```diff
@@ -1,31 +1,44 @@
-*** Settings ***		
-
-Resource        robot/Cumulus/resources/NPSP.robot	
+*** Settings ***
+Resource        robot/Cumulus/resources/NPSP.robot
 Library         cumulusci.robotframework.PageObjects
-...             robot/Cumulus/resources/ContactPageObject.py	
-Suite Setup     Open Test Browser		
-Suite Teardown  Delete Records and Close Browser		
+...             robot/Cumulus/resources/ContactPageObject.py
+...             robot/Cumulus/resources/AccountPageObject.py
+...             robot/Cumulus/resources/OpportunityPageObject.py
+...             robot/Cumulus/resources/NPSP.py
+Suite Setup     Run keywords
+...             Open Test Browser
+...             Setup Test Data
+Suite Teardown  Delete Records and Close Browser
+
+***Keywords***
+# Setup a contact with parameters specified
+Setup Test Data
+    Setupdata   contact   ${contact1_fields}
 
+*** Variables ***
+&{contact1_fields}  Email=test@example.com
 
 *** Test Cases ***
 
 Create Open Recurring Donation With Monthly Installment
     [Documentation]              This test verifies that a Recurring Donation can be created through the UI.
-    &{contact} =                 API Create Contact    Email=skristem@robot.com		
-    Go To Page                   Details    Contact    object_id=&{contact}[Id]		
-    Click More Actions Button		
-    Click Link                   link=New Open Recurring Donation		
+
+    Go To Page                           Details
+    ...                                  Contact
+    ...                                  object_id=${data}[contact][Id]
```

Comment:
You are creating a dictionary instead of using an API to create contact? Can I know what is the reason behind it?